stablehlo/dialect/StablehloOps.cpp

```diff
@@ -1945,7 +1959,6 @@ LogicalResult AllToAllOp::inferReturnTypeComponents(
 
```

Comment:
Similarly to https://github.com/openxla/stablehlo/pull/503/files#r1032745628, the `splitDimension < 0` and `concatDimension < 0` checks should be above the `if (!operandRankedType) { ... }` check, i.e. we should check that even for unranked operand types.